phenotrex/structure/records.py

```diff
@@ -30,20 +34,23 @@ def __repr__(self):
 
 @dataclass
 class GroupRecord:
-    """ TODO add docstring """
+    """
+    Structure containing grouping information for each sample for Leave-one-group-out CV.
+    """
     identifier: str
-    group_name: str
-    group_id: int
+    group_name: Optional[str]
+    group_id: Optional[int]
 
     def __repr__(self):
         return f"{self.identifier} group({self.group_name})={self.group_id}"
```

Comment:
When two attributes are optional, the repr will probably not look very nice.

Reply:
the GroupRecord is in any case not supposed to be used outside of a training record so I modified the repr of TrainingRecord to ensure it looks okay if the group record is empty.